scp/scpGenOperatorInterpreter.hpp

```diff
@@ -1,31 +1,36 @@
 /*
-* This source file is part of an OSTIS project. For the latest info, see http://ostis.net
-* Distributed under the MIT License
-* (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
-*/
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
 
 #pragma once
 
-#include "sc-memory/sc_addr.hpp"
-#include "sc-memory/sc_object.hpp"
-#include "sc-memory/kpm/sc_agent.hpp"
-#include "scpKeynodes.hpp"
+#include "scpOperator.hpp"
 
-#include "scpGenOperatorInterpreter.generated.hpp"
+#include <sc-memory/sc_agent.hpp>
 
 namespace scp
 {
 
-class ASCPGenOperatorInterpreter : public ScAgent
+class ASCPGenOperatorInterpreter : public ScAgent<ScEventAfterGenerateOutgoingArc<ScType::EdgeAccessConstPosPerm>>
 {
-    SC_CLASS(Agent, Event(Keynodes::active_action, ScEvent::Type::AddOutputEdge))
-    SC_GENERATED_BODY()
-
 public:
-    SC_PROPERTY(Keynode("sc_agent_of_scp_operator_gen_interpreting"), ForceCreate)
-    static ScAddr msAgentKeynode;
+  ScAddr GetActionClass() const override;
 
-};
+  ScResult DoProgram(ScEventAfterGenerateOutgoingArc<ScType::EdgeAccessConstPosPerm> const & event, ScAction & action)
+      override;
+
+  ScAddr GetEventSubscriptionElement() const override;
+
+  bool CheckInitiationCondition(ScEventAfterGenerateOutgoingArc<ScType::EdgeAccessConstPosPerm> const & event) override;
 
-}
+  static inline ScKeynode const msAgentKeynode{"sc_agent_of_scp_operator_gen_interpreting"};
```

Comment:
Why did you leave these ones, if they are don't used? Use them

Reply:
I don't know what is their purpose so I don't want to remove them